### ISSUE TITLE
Invalid form data for user are stored in token

### ISSUE THREAD

**Issue by Shoplifter** [symfony/symfony#26194]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | ?
| RFC?             | no
| Symfony version  | 4.0.4

I am using Guard and an Authenticator extending the AbstractFormLoginAuthenticator.
I am logged in and edit my own user account in an user administration form and submit.
Some constraint fails (e.g. a required field left blank)  - I get the form with error message, but am logged out. Therefore the next request redirects me to the login page.

I used the (almost) same code in a former project with Symfony 3.3 where this bug did not occur.
(only difference now is the new supports method to be implemented, which former was a part of getCredentials)

**Comment by Simperfit** [comment]
Could you please provide a reproducer ? It could help us alot with understanding the issue.

**Comment by iltar** [comment]
https://stovepipe.systems/post/decoupling-your-security-user
https://stovepipe.systems/post/avoiding-entities-in-forms

**Comment by xabbuh** [comment]
Can you check if this is related (the same as) #25992?

**Comment by iltar** [comment]
This could be another cause: https://github.com/symfony/symfony/pull/23882

**Comment by MathieuMindid** [comment]
I have the same issue with a sample way to test, i publish a code to explain : https://github.com/MathieuMindid/symfony_bug26194

You have a simple User entity with id, username and password. You can load to users with the fixtures (`make fixtures`)

Login with the first user (login : john.doe / pass : john.doe) then you can edit the username as you want all works.
then if tou try to edit with non-unique value : "m-smith" you will get an error (it's normal i defined the fields as unique in my entities), and on the next update of the form you will get logout.

**Comment by Shoplifter** [comment]
@Simperfit You still need a reproducer? I think MathieuMindid already provided one.
@MathieuMindid Thank you very much for doing so.

**Comment by Simperfit** [comment]
Thanks for the reproducer. I'll try too look into it and to confirm it too.

**Comment by xabbuh** [comment]
The point is that you save the user entity as the security user in the session. Now when you submit the form the user object is changed (in memory even if it isn't persisted). This means that the username in the session was changed. On the following request the username from the session is compared to the one from the database and now they differ since only the username in the session was changed. Consequently the user is logged out as the listener detects the user as being changed.

Basically @iltar provided the solution for this problem in https://github.com/symfony/symfony/issues/26194#issuecomment-366200414.
